dsp/modules/googlevertexai.py

```diff
@@ -0,0 +1,174 @@
+"""Module for interacting with Google Vertex AI."""
+from typing import Any, Dict
+
+import backoff
+from pydantic_core import PydanticCustomError
+
+from dsp.modules.lm import LM
+
+try:
+    import vertexai  # type: ignore[import-untyped]
+    from vertexai.language_models import CodeGenerationModel, TextGenerationModel
+    from vertexai.preview.generative_models import GenerativeModel
+except ImportError:
+    pass
+
+
+def backoff_hdlr(details):
+    """Handler from https://pypi.org/project/backoff/"""
+    print(f"Backing off {details['wait']:0.1f} seconds after {details['tries']} tries "
+          f"calling function {details['target']} with kwargs "
+          f"{details['kwargs']}")
+
+
+def giveup_hdlr(details):
+    """wrapper function that decides when to give up on retry"""
+    if "rate limits" in details.message:
+        return False
+    return True
+
+class GoogleVertexAI(LM):
+    """Wrapper around GoogleVertexAI's API.
+
+    Currently supported models include `gemini-pro-1.0`.
+    """
+
+    def __init__(
+        self, model_name: str = "text-bison@002", **kwargs,
+    ):
+        """
+        Parameters
+        ----------
+        model : str
+            Which pre-trained model from Google to use?
+            Choices are [`text-bison@002`]
```

Comment:
can you clarify what the supported models are? Additionally, `model` is not a parameter. `model_name` is. This was confusing when I was using this code for a small demo project.